python/tskit/drawing.py

```diff
@@ -679,33 +684,90 @@ def __init__(
         canvas_size=None,
     ):
         """
-        Creates self.drawing, an svgwrite.Drawing object for further use, and populates
-        it with a base group. The root_groups will be populated with
-        items that can be accessed from the outside, such as the plotbox, axes, etc.
+        Creates root XML element and base groups
         """
-
         if root_svg_attributes is None:
             root_svg_attributes = {}
         if canvas_size is None:
             canvas_size = size
-        dwg = svgwrite.Drawing(size=canvas_size, debug=True, **root_svg_attributes)
 
+        # Create SVG root element
         self.image_size = size
         self.plotbox = Plotbox(size)
         self.root_groups = {}
         self.svg_class = svg_class
         self.timescaling = None
         self.root_svg_attributes = root_svg_attributes
-        self.dwg_base = dwg.add(dwg.g(class_=svg_class))
-        self.drawing = dwg
+
+        # Create SVG document
+        svg_attrs = {
```

Comment:
This is quite low-level directly working with SVG. How about we add a new class "SvgWriter" or something that abstracts things a bit, and gives us the flexibility to switch backend?

Reply:
I've had a quick stab in the last commit, seems promising! Just a few details to fix.